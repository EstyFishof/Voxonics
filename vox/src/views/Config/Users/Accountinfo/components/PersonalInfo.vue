<template>
  <div class="personal-info-wrap" :class="{'non-pointer-events': roleAgent === 'agent'}">
    <div class="wrap-body">
      <span class="file-loader-span">
        <FileLoader :userUidProp="true" />
      </span>
      <div class="personal-info account-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name*
              <input
                @input="changeUserInfo({ name: userInfo.name }); checkName()"
                placeholder="Enter user name"
                class="input-field"
                maxlength="16"
                v-model.trim="userInfo.name"
                :class="{ error: !userInfo.name.length }"
              />
            </label>
            <div v-if="errors.name" class="alert">{{ errors.name }}</div>
          </div>
          <div class="item-container with-err">
            <label
              >Phone
              <input
                placeholder="Only digits"
                class="input-field"
                type="text"
                v-model.trim="userInfo.phone"
                maxlength="15"
                @input="changeUserInfo({ phone: userInfo.phone || null }); checkPhone()"
              />
            </label>
            <div v-if="errors.phone" class="alert">{{ errors.phone }}</div>
          </div>
          <div class="item-container">
            <label class="options-label" ref="list" v-click-outside="hide"
              >Location:
              <input
                class="input-field num"
                type="text"
                @input="search($event.target.value)"
                v-model="userInfo.location"
                placeholder="London, UK"
                maxlength="16"
              />
              <span
                @click="setParam('showList', 'location', 'location')"
                class="triangle"
                :class="{ 'rotate-arrow': showList === 'location' }"
              >
                <img src="@/assets/img/icons/select-arrow.svg" alt="select" />
              </span>
              <DropdownList v-show="showList === 'location'" :dataList="getGlobalLocations"
                @clickFunc="(item) => {setParam('userInfo.location', item.location); changeUserInfo({ location: userInfo.location || null }) }" 
                :clickFuncAll="null" :classNameList="'list-related-numbers'"  @scrollFunc="(event) => funScroll(event)" :itemText="(item) => item.location" 
                :classNameItemComputed="(item) => userInfoVuex.location === item.location ? 'selected' : ''" />
            </label>
          </div>
          <div class="item-container with-err">
            <label
              >Agent Number (id)*
              <input
                placeholder="000"
                class="input-field"
                type="text"
                v-facade="'###'"
                @input="changeUserInfo({ internalNumber: userInfo.internalNumber }); checkInternalNumber()"
                v-model.trim="userInfo.internalNumber"
                :class="{ error: errors.internalNumber }"
                maxlength="3"
              />
            </label>
            <div v-if="errors.internalNumber" class="alert">
              {{ errors.internalNumber }}
            </div>
          </div>
          <div class="item-container with-err">
            <label
              >Email
              <input
                @input="changeUserInfo({ email: userInfo.email || null }); checkEmail()"
                placeholder="example@voxonics.com"
                class="input-field"
                type="email"
                maxlength="36"
                v-model.trim="userInfo.email"
              />
            </label>
            <div v-if="errors.email" class="alert">{{ errors.email }}</div>
          </div>
          <div class="personal-info__second-part">
            <div class="item-container">
              <label
                >Position
                <input
                  @input="changeUserInfo({ position: userInfo.position || null })"
                  placeholder="Sales Manager"
                  class="input-field"
                  type="text"
                  maxlength="16"
                  v-model.trim="userInfo.position"
                />
              </label>
            </div>
          </div>
          <div class="hr"></div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import FileLoader from "@/components/FileLoader"
import DropdownList from '@/components/DropdownList'
import Validate from "@/lib/validate"
import { mapGetters } from "vuex"
import Api from "@/API/Global.api.js"
import store from "../../../../../store"
import _ from "lodash"
import ClickOutside from "@/lib/clickDirective"

export default {
  name: "PersonalInfo",
  components: {
    FileLoader,
    DropdownList
  },
  data() {
    return {
      unsubscribe: () => {},
      userInfo: {
        name: "",
        phone: "",
        location: "",
        position: "",
        email: "",
        internalNumber: ""
      },
      errors: {
        name: "",
        phone: "",
        email: "",
        internalNumber: "",
        password: "",
        confirm: ""
      },
      count: 10,
      showList: "",
      perPage: 9,
      timer: null
    }
  },
  computed: {
    ...mapGetters(["userInfoVuex", "userInfoComponent", "roleAgent", "getGlobalLocations"])
  },
  methods: {
    funScroll({ target }) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        if (this.count > this.perPage) {
          this.perPage += 9
          this.fetch("location", this.perPage, this.userInfo.location)
        }
      }
    },
    search(paramSearch) {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch("location", this.perPage, paramSearch), 800)
    },
    checkName() {
      Validate.checkName(this)
    },
    checkPhone() {
      Validate.checkPhone(this)
    },
    checkEmail() {
      Validate.checkEmail(this)
    },
    checkInternalNumber() {
      const num = window.location.href.split("/").pop()
      Validate.checkInternalNumber(this, num)
    },
    checkPassword() {
      Validate.checkPassword(this)
    },
    confirmPassword() {
      Validate.confirmPassword(this)
    },
    fetch(fetchParam, perPage, value) {
      this.$store.dispatch("setSpinner", true)
      switch (fetchParam) {
        case "location":
          return Api.getLocation(perPage, value).then(data => {
            this.count = data.paginator.count
            return data.data
          })
        default:
          break
      }
    },
    setParam(param, value, fetchParam) {
      if (_.get(this, param) === value) {
        _.set(this, param, "")
      } else {
        _.set(this, param, value)
      }
      if (fetchParam) {
        this.fetch(fetchParam, this.perPage, this.userInfo.location)
      }
    },
    changeUserInfo(info) {
      this.showList = ""
      if (this.userInfoComponent.permission.role !== "agent") {
        this.checkInternalNumber()
        this.$store.dispatch("changeUserInfo", info)
      }
    },
    hide() {
      this.showList = ""
    }
  },
  mounted() {
    if (this.roleAgent !== "agent") {
      //Если есть записи в сторе, переносим их в компонент
      for (const key in this.userInfoVuex) {
        this.userInfo[key] = this.userInfoVuex[key]
      }
      this.unsubscribe = store.subscribe(mutation => {
        if (mutation.type === "setUserInfo") {
          for (const key in this.userInfoVuex) {
            this.userInfo[key] = this.userInfoVuex[key]
          }
        }
      })
    } else {
      for (const key in this.userInfoComponent) {
        this.userInfo[key] = this.userInfoComponent[key]
      }
    }
  },
  beforeDestroy() {
    this.unsubscribe()
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss">
.personal-info-wrap {
  background-color: var(--substrate);
  border-radius: 8px 8px 0 0;
  // border: 2px solid var(--cdr-border-row);
  border-bottom: none;

  &.non-pointer-events {
    pointer-events: none;
  }

  .wrap-body {
    display: flex;
    flex-direction: row;
    padding: 0 0 36px 36px;

    .file-loader-span {
      margin: 24px;
    }

    .personal-info {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      justify-content: space-between;
      align-items: center;
      margin-left: 36px;

      .hr {
        width: 100%;
        height: 2px;
        background: var(--cdr-border-row);
        border-radius: 2px;
        flex-grow: 1;
        margin: 32px 24px 0 24px;
      }

      &__first-part,
      &__second-part {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        align-content: flex-start;
        justify-content: flex-start;

        .item-container {
          margin: 16px 24px;
          position: relative;

          label {
            font-size: 18px;
            font-weight: 700;
            text-align: left;
            color: var(--light-blue);
          }

          .error {
            border: 1px solid red;
          }

          .alert {
            position: absolute;
            font-size: 16px;
            font-weight: 300;
            color: #ff0000;
            bottom: -24px;
          }

          .options-label {
            .input-field {
              margin-top: 12px;
              display: block;
              padding: 5px 16px;
              width: 292px;
              height: 35px;
              background: var(--input-bg);
              box-shadow: 0 8px 16px rgba(0, 0, 0, 0.12);
              border-radius: 4px;
              color: var(--font-panel);
              border: none;
            }

            .triangle {
              height: 35px;
              width: calc(100% - 7px);
            }

            .list-related-numbers {
              max-height: 200px;
              overflow-y: auto;
            }
          }
        }
      }
    }
  }

  .with-err {
    position: relative;
    margin-bottom: 30px;
  }
}
@media (max-width: 1408px) {
  .input-field {
    width: 270px;
  }
}

</style>
<style lang="scss">
@media (max-width: 1408px) {
  .personal-info-wrap {
    .wrap-body {
      .account-info {
        .personal-info {
          &__first-part {
            .item-container {
              .options-label {
                .input-field {
                  width: 270px !important;
                }
              }
            }
          }
        }
      }
    }
  }
}
</style>
