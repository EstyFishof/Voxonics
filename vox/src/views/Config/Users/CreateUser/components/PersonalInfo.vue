<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2 class="wrap-head-title">PERSONAL INFO</h2>
      <span class="hr"></span>
    </div>

    <div class="wrap-body">
      <FileLoader :userUidProp="false" />
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <label>
              Name*
              <input
                placeholder="Enter user name"
                class="input-field"
                @input="setUserInfo({ name: userInfo.name }); checkName()"
                maxlength="16"
                v-model.trim="userInfo.name"
                required
                :class="{ error: requiredFields.name }"
              />
            </label>
            <div v-if="errors.name" class="alert">{{ errors.name }}</div>
          </div>
          <div class="item-container with-err">
            <label
              >Phone
              <input
                placeholder="Only digits"
                class="input-field"
                type="text"
                v-model.trim="userInfo.phone"
                maxlength="15"
                @input="setUserInfo({ phone: userInfo.phone }); checkPhone()"
              />
            </label>
            <div v-if="errors.phone" class="alert">{{ errors.phone }}</div>
          </div>
          <div class="item-container">
            <label class="options-label" ref="list" v-click-outside="hide"
              >Location:
              <input
                class="input-field num"
                type="text"
                @input="search($event.target.value)"
                v-model="userInfo.location"
                placeholder="London, UK"
                maxlength="16"
              />
              <span
                @click="setParam('showList', 'location', 'location')"
                class="triangle"
                :class="{ 'rotate-arrow': showList === 'location' }"
              >
                <img src="@/assets/img/icons/select-arrow.svg" alt="select" />
              </span>
              <DropdownList v-show="showList === 'location'" :dataList="getGlobalLocations"
                @clickFunc="(item) => {setUserInfo({ location: item.location }); setParam('userInfo.location', item.location)}" 
                :clickFuncAll="null" :classNameList="'list-related-numbers'"  @scrollFunc="(event) => funScroll(event)" :itemText="(item) => item.location" 
                :classNameItemComputed="(item) => getForm.location === item.location ? 'selected' : ''" />
            </label>
          </div>
          <div class="item-container with-err">
            <label
              >Agent Number (id)*
              <input
                placeholder="000"
                class="input-field"
                type="text"
                v-facade="'###'"
                @input="setUserInfo({ internalNumber: userInfo.internalNumber }); checkInternalNumber()"
                v-model.trim="userInfo.internalNumber"
                required
                :class="{ error: requiredFields.internalNumber }"
                maxlength="3"
              />
            </label>
            <div v-if="errors.internalNumber" class="alert">
              {{ errors.internalNumber }}
            </div>
          </div>
          <div class="item-container with-err">
            <label
              >Email
              <input
                placeholder="example@voxonics.com"
                class="input-field"
                type="email"
                @input="setUserInfo({ email: userInfo.email }); checkEmail()"
                maxlength="36"
                v-model.trim="userInfo.email"
              />
            </label>
            <div v-if="errors.email" class="alert">{{ errors.email }}</div>
          </div>
          <div class="item-container">
            <label
              >Position
              <input
                @input="setUserInfo({ position: userInfo.position })"
                placeholder="Sales Manager"
                class="input-field"
                type="text"
                maxlength="16"
                v-model.trim="userInfo.position"
              />
            </label>
          </div>
                    <div class="item-container password-field">
            <label
              >Password*
              <div
                class="icon-eye"
                :class="{
                  'icon-eye': passwordFieldType === 'text',
                  'icon-eye-close': passwordFieldType === 'password'
                }"
                @click="switchVisibility"
              ></div>
              <input
                placeholder="Enter password"
                class="input-field"
                :type="passwordFieldType"
                @input="setUserInfo({ password: userInfo.password }); checkPassword()"
                v-model.trim="userInfo.password"
                required
                :class="{ error: requiredFields.password }"
                maxlength="16"
              />
            </label>

            <div v-if="errors.password" class="alert-pass">
              {{ errors.password }}
            </div>
          </div>
          <div class="item-container password-field">
            <label
              >Confirm*
              <div v-if="switchVisibilityConfirm" @click="switchVisibilityConfirm">
                <span
                  :class="{
                    'icon-eye': passwordConfirmFieldType === 'text',
                    'icon-eye-close': passwordConfirmFieldType === 'password'
                  }"
                ></span>
              </div>
              <input
                placeholder="Confirm password"
                class="input-field"
                :type="passwordConfirmFieldType"
                v-model.trim="userInfo.confirm"
                @input="setUserInfo({ confirm: userInfo.confirm }); confirmPassword()"
                required
                :class="{ error: requiredFields.password }"
                maxlength="16"
              />
              <div v-if="errors.confirm" class="alert-pass">
                {{ errors.confirm }}
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import FileLoader from "@/components/FileLoader"
import DropdownList from '@/components/DropdownList'
import Validate from "@/lib/validate"
import Api from "@/API/Global.api.js"
import ClickOutside from "@/lib/clickDirective"
import { mapGetters } from "vuex"
import _ from "lodash"

export default {
  name: "PersonalInfo",
  components: {
    FileLoader,
    DropdownList
  },
  props: ["requiredFields"],
  data() {
    return {
      userInfo: {
        name: "",
        phone: "",
        location: "",
        position: "",
        email: "",
        internalNumber: "",
        password: "",
        confirm: ""
      },

      errors: {
        name: "",
        phone: "",
        email: "",
        internalNumber: "",
        password: "",
        confirm: ""
      },

      passwordFieldType: "password",
      passwordConfirmFieldType: "password",
      passwordConfirmIcon: "icon-eye",

      showList: "",
      perPage: 9,
      timer: null
    }
  },
  computed: {
    ...mapGetters(["getForm", "getGlobalLocations"])
  },
  methods: {
    funScroll({ target }) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        this.perPage += 9
        this.fetch("location", this.perPage, this.userInfo.location)
      }
    },
    search(paramSearch) {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch("location", this.perPage, paramSearch), 800)
    },
    checkName() {
      Validate.checkName(this)
    },
    checkPhone() {
      Validate.checkPhone(this)
    },
    checkEmail() {
      Validate.checkEmail(this)
    },
    checkInternalNumber() {
      Validate.checkInternalNumber(this)
    },
    checkPassword() {
      Validate.checkPassword(this)
    },
    confirmPassword() {
      Validate.confirmPassword(this)
    },
    switchVisibility() {
      this.passwordFieldType = this.passwordFieldType === "password" ? "text" : "password"
    },
    switchVisibilityConfirm() {
      this.passwordConfirmFieldType = this.passwordConfirmFieldType === "password" ? "text" : "password"
      this.passwordConfirmIcon = this.passwordConfirmIcon === "icon-eye" ? "icon-eye-close" : "icon-eye"
    },
    fetch(fetchParam, perPage, value) {
      this.$store.dispatch("setSpinner", true)
      switch (fetchParam) {
        case "location":
          Api.getLocation(perPage, value)
          break
        default:
          break
      }
    },
    setParam(param, value, fetchParam) {
      if (_.get(this, param) === value) {
        _.set(this, param, "")
      } else {
        _.set(this, param, value)
      }
      if (fetchParam) {
        this.fetch(fetchParam, this.perPage, this.userInfo.location)
      }
    },
    setUserInfo(info) {
      this.$store.dispatch("setUserInfo", info)
      this.showList = ""
    },
    hide() {
      this.showList = ""
    }
  },
  mounted() {
    //Если есть записи в сторе, переносим их в компонент
    for (const key in this.getForm) {
      if (
        Object.prototype.hasOwnProperty.call(this.userInfo, key) &&
        Object.prototype.hasOwnProperty.call(this.getForm, key)
      ) {
        this.userInfo[key] = this.getForm[key]
      }
    }
    this.popupItem = this.$refs.list
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/users/personalInfo.scss"></style>
<style lang="scss" scoped>
@media (max-width: 1408px) {
  .personal-info-wrap {
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          margin: 16px 22px;
        }
      }
    }
  }
}
</style>
