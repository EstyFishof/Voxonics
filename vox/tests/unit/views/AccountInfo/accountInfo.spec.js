import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import InputFacade from 'vue-input-facade';
import AccountInfo from '@/views/Config/Users/Accountinfo/Accountinfo';
import PersonalInfo from '@/views/Config/Users/Accountinfo/components/PersonalInfo';
import user from "../EditUser/__mock__/user";
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
localVue.directive('facade', {
    inserted: function (el) {
        const e = el;
    }
});
const router = new VueRouter();

describe('edit user', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters;
    let store;

    beforeEach(() => {
        getters = {
            userInfoVuex: () => user,
            userUid: () => '',
            departments: () => [],
            userInfo: () => user,
            changeFields: () => [],
            userInfoComponent: () => {{'agent';}},
            roleAgent: () => 'manager'
        };
        store = new Vuex.Store({
            getters
        });
    });
    it('should  Vue instance', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('vuex defined', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        expect(wrapper.vm.userInfoVuex).toBeDefined();
        expect(wrapper.vm.userUid).toBeDefined();
        expect(wrapper.vm.departments).toBeDefined();
    });
    it('data defined', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        expect(wrapper.vm.showOptions).toBeDefined();
        expect(wrapper.vm.array).toBeDefined();
        expect(wrapper.vm.edit).toBeDefined();
    });
    it('set show options', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'setShowOptions');
        wrapper.vm.setShowOptions('blabla');
        expect(wrapper.vm.showOptions).toEqual('blabla');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('clearUser', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'clearUser');
        wrapper.vm.clearUser();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('handlerSubmit', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('switchMode', () => {
        const wrapper = shallowMount(AccountInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'switchMode');
        wrapper.vm.switchMode();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

});

describe('personal info in AccountInfo', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters, actions;
    let store;
    const mockUserInfoComponent = {
        permission: {
            role: 'agent'
        }
    };
    beforeEach(() => {
        getters = {
            userInfoVuex: () => user,
            userInfo: () => user,
            userInfoComponent: () => mockUserInfoComponent,
            roleAgent: () => 'manager'
        };
        actions = {
            changeUserInfo: jest.fn()
        };
        store = new Vuex.Store({
            actions,
            getters
        });
    });
    it('should Vue instance', () => {
        const wrapper = shallowMount(PersonalInfo, {
            localVue, router, store, directives: {
                facade: el => {
                }
            }
        });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('userInfoVuex defined', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        expect(wrapper.vm.userInfoVuex).toBeDefined();
    });
    it('checkName ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkName');
        wrapper.vm.checkName(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkPhone', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkPhone');
        wrapper.vm.checkPhone(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkEmail', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkEmail');
        wrapper.vm.checkEmail(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('checkInternalNumber ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'checkInternalNumber');
        wrapper.vm.checkInternalNumber(wrapper.vm);
        expect(spy).toHaveBeenCalled();
    });
    it('changeUserInfo ', () => {
        const wrapper = shallowMount(PersonalInfo, { localVue, router, store });
        const spy = jest.spyOn(wrapper.vm, 'changeUserInfo');
        wrapper.vm.changeUserInfo('info');
        expect(spy).toHaveBeenCalled();
    });
});
