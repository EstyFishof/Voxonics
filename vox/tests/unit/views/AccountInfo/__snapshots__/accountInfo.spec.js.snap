// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit user Check snapshot 1`] = `
<div
  class="wrap-main"
>
  <div
    class="wrap-form account-wrap"
  >
    <form>
      <personalinfo-stub />
       
      <div
        class="bottom-field"
      >
        <!---->
         
        <div
          class="tabs"
        >
          <div
            class="left-tabs"
          >
            <div
              class="tab active"
            >
              
              Permissions
            
            </div>
             
            <div
              class="tab"
            >
              
              Departments
            
            </div>
             
            <div
              class="tab queues"
            >
              Queues
              
              <ul
                class="secondary-list"
                style="display: none;"
              >
                <li
                  class="secondary-list__item"
                >
                  Using
                
                </li>
                 
                <li
                  class="secondary-list__item"
                >
                  Managing
                
                </li>
              </ul>
            </div>
             
            <div
              class="tab"
            >
              Settings overview
            </div>
          </div>
           
          <div
            class="right-tabs"
          >
            <controlpanel-stub
              apply="function () { [native code] }"
              discard="function () { [native code] }"
            />
          </div>
        </div>
         
        <div
          class="body"
        >
          <permission-stub />
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`personal info in AccountInfo Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-body"
  >
    <fileloader-stub />
     
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label>
            
            Name*
            
            <input
              class="input-field error"
              maxlength="32"
              placeholder="Enter user name"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Phone
            
            <input
              class="input-field"
              maxlength="15"
              placeholder="Only digits"
              type="text"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container"
        >
          <label>
            Location
            
            <input
              class="input-field"
              placeholder="London, UK"
              type="text"
            />
          </label>
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Agent Number (id)*
            
            <input
              class="input-field"
              placeholder="000"
              type="text"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="item-container with-err"
        >
          <label>
            Email
            
            <input
              class="input-field"
              maxlength="36"
              placeholder="example@voxonics.com"
              type="email"
            />
          </label>
           
          <!---->
        </div>
         
        <div
          class="personal-info__second-part"
        >
          <div
            class="item-container"
          >
            <label>
              Position
              
              <input
                class="input-field"
                maxlength="16"
                placeholder="Sales Manager"
                type="text"
              />
            </label>
          </div>
        </div>
         
        <div
          class="hr"
        />
      </div>
    </div>
  </div>
</div>
`;
